---
title: "[Working title]: Foliar fungal symbionts in sympatric yellow monkeyflowers along elevation gradients in the Sierra Nevada"
date: last-modified
date-format: "MMMM D, YYYY"
bibliography: references.bib
csl: apa.csl
author:
  - name: Bolívar Aponte Rolón
    orcid: 0000-0002-2544-4551
    email: baponterolo@tulane.edu
    corresponding: true
    affiliations: 
      - ref: tu
        name: Tulane University
        department: Ecology and Evolutionary Biology
        city: New Orleans
        state: LA
        country: US
        url: www.tulane.edu
  - name: Kathleen G. Ferris
    email: kferris@tulane.edu
    corresponding: false
    affiliations: 
      - ref: tu
  - name: Sunshine A. Van Bael
    orcid: 0000-0001-7317-3533
    email: svanbael@tulane.edu
    corresponding: false
    affiliation: 
      - ref: tu
      
affiliations:
  - id: tu
    name: Department of Ecology and Evolutionary Biology, Tulane University, 6823 St. Charles Avenue, New Orleans, LA 70118

filters:
  - authors-block

format:
  # pdf:
  #   documentclass: scrartcl
  #   include-in-header:
  #       text: '\usepackage{lineno}\linenumbers'
  #   linestretch: 2.0
  #   margin-left: 1in
  #   margin-right: 1in
  #   margin-top: 1in
  #   margin-bottom: 1in
  #   mainfont: TeX Gyre Termes #Alternative to Times New Roman
  #   sansfont: TeX Gyre Termes #Another option is Liberation Serif
  #   number-sections: true
  #   citation-package: biblatex
  #   #keep_tex: true
  #   keep_md: true
  docx:
    reference-doc: custom-reference-doc.docx
    number-sections: true
    tbl-colwidths: true
    #linestretch: 2.0
    citation-package: biblatex
crossref:
  custom:
    - kind: float
      key: suppfig
      latex-env: suppfig
      reference-prefix: Fig. S
      space-before-numbering: false
      latex-list-of-description: Supplementary Figure
    
editor: source
execute:
  echo: false
warning: false
---

## Keywords

# Abstract

# Introduction

Phenotypic plasticity and genetic adaptation are key drivers of evolution, and both processes may
aid in plant adaptation to a warming climate [@jump2005]. Plant’s colonization of
land across multiples climates has been aided by their parallel evolution with fungi and bacteria;
however, little is known about how plant microbiomes influence or are influenced by local
adaptation or phenotypic plasticity of plants. Lack of knowledge about this is a problem because
ecology aims to predict how plant distribution may change as the climate becomes further
altered. Previous studies have suggested that foliar endophytic fungi (FEF) can alter plant traits
under stressful conditions such as drought [@song2016]. FEF live in plant leaves and they
may contribute to local adaptation and/or phenotypic plasticity.
Host genetics may shape the foliar microbiomes of plants. For example, a study on the classical
plant model, Arabidopsis thaliana, revealed that bacterial and fungal communities in leaves are
shaped by host genotype, at least for the most common OTUs [@horton2014]. Another
ideal model system to explore the interactions between leaf traits and FEF are the yellow
monkeyflowers in the *Mimulus guttatus* species complex. Over the last 2 decades, the M.
guttatus species complex has become a model system for answering questions about local
adaptation and phenotypic plasticity [@wu2008; @yuan2019]. Experimental manipulation
is relatively easy due to its small genome (about 430 Mb), short generation time, high fecundity,
self-compatibility, and ease of greenhouse propagation [@society2019; @wu2008]
Currently, studies of the M. guttatus species complex focus on the genetic and environmental
factors driving local adaptation and phenotypic plasticity. The evolution of leaf shape for 3
edaphic species in the M. guttatus complex is controlled by overlapping genetic regions [@ferris2015]. Ferris et al., [-@ferris2015] points to multiple events of leaf shape evolution in the *M. guttatus* species complex and their associated habitats. Recent work has narrowed the gene architecture at play in incomplete reproductively isolated sympatric populations of *Mimulus laciniatus* and *Mimulus guttatus*, shedding light on the location of quantitative trait loci (QTL) for the first flower node, flowering time, corolla width, corolla length, and leaf shape [@ferris2017] and suggesting that large-effect loci are likely experiencing gene-flow in these populations. There is scarce research that focuses on the symbionts of Mimulus spp. Beslile et al. [-@beslile2012] reported on the distribution of diverse fungal communities in the flower nectar of *Mimulus aurantiacus*. The authors considered flowers as islands in a metapopulation system and found that the frequency of micro fungi (i.e., yeasts) was significantly correlated to the location of the plant and marginally
correlated with the density of the flowers in the plant [@beslile2012]. To our knowledge,
no previous studies have considered how locally adapted or phenotypic traits influence *in planta* symbioses and, in turn, how these might play a role in the development of such traits.


### Hypothesis and Prediction
#### Hypothesis

Environmental factors such as elevation and geographic distance have can influence local
adaptation, phenotypic plasticity of leaf traits, as well as symbiotic FEF communities in sympatric populations of *M. guttatus*, *M. laciniatus* , and *M. nasutus*.

#### Justification. 
There is little information on the potential symbiotic relationships the M. guttatus
species complex may host. Because of this, it is valuable to create a detailed account of the
endophyte communities present in the Mimulus guttatus species complex along an environmental
gradient (e.g., elevation).

#### Prediction. 
The diversity, richness, and abundance of FEF communities will be similar within
the same sites (alpha diversity) than among sites (beta diversity) regardless
of host species.


^ NEEDS WORK
<!-- In answering my main questions for this research project I aimed to create a detailed account of the endophyte communities present in *M. guttatus* and *M. laciniatus* along an environmental gradient (e.g., elevation) to capture variation in population habitat, genetics and phenotypes and their associated endophyte communities. Ultimately, shedding light on how local adaptations, phenotypic traits, and endophyte communities in the *M. guttatus* complex interact. -->

# Materials and Methods

## Sample Collection

We collected plant specimens populations of *M. guttatus*, *M. laciniatus*, and *M. nasutus* (syn. *Erythranthe guttata* , *Erythranthe laciniata*, and *Erythranthe nasuta*) across Stanislaus National Forest (SNF), Sierra National Forest (SINF) and Yosemite National Park (YNP), CA, USA. during We haphazardly selected sites close to the main roads based on the presence of a viable population of at least \~ 50 individuals per species. Samples collected from YNP were collected from non-wilderness areas on the side of the road. We determined population viability ensuring that they had individuals flowering or close to flowering stage. We collected between 6 - 12 individuals per species per site. We selected individuals that possessed healthy looking leaves, no visible signs of pathogen damage or senescence. At sites were two species were present we collected individuals that were at least \~ 25 meters apart. We collected sample specimens by carefully uprooting the plant and placing into individual plastic bags (e.g., Ziploc®) and preserving in an ice chest until return to the field laboratory at the UC Merced Yosemite Field Station, YNP, CA, USA. Plant specimens were processed within 8 hrs of collection.

## Leaf traits measurement

From each plant, we measured leaf traits: leaf thickness (LT), leaf punch strength (LPS), leaf mass per area (LMA), anthocyanin content index (ACI) which are known to be associated with FEF communities [@tellezTraits2022] as well as leaf lobe index (LBI) [@ferris2015]. We cleaned plants with tap water to remove all soil and debris remnants from the leaves and roots. We removed all healthy leaves (\~ 5 - 10) from the stems and took three measurements per trait from three haphazardly selected leaves from individual plants, with the exception of LBI, only one leaf per plant. We used a transparency film to hold the leaf in place and flatten, after which we took a digital photograph for analysis in ImageJ [v1.52r; @schneider2012]. To calculate the LBI, we followed [-@ferris2015]. Leaf lobing is calculated as the convex hull area minus the true leaf area divided by convex hull area. We measured ACI content with ACM-200plus (Opti-Sciences Inc. Hudson, New Hampshire, U.S.A.) on haphazardly selected locations of the leaf surface (working from the petiole out to the leaf tip) [@tellezTraits2022]. The ACM-200 calculates an ACI value from the ratio of % transmittance at 931 nm/% transmittance at 525 nm [@opti-sciencesinc], effectively accounting for leaf thickness. We measured LT (μm) with a Mitutoyo 7327 Micrometer Gauge (Mitutoyo, Takatsu-ku, Kawasaki, Japan) on haphazardly selected locations of the leaf lamina, taking care to avoid major and secondary veins. We used an Imada DST-11a digital force gauge (Imada Inc., Northbrook, IL, United States) to measure LPS, a measure of leaf toughness, on the lamina of each leaf selected, avoiding minor leaf veins when possible [@tellezTraits2022]. It functions by conducting punch-and-die tests with a sharp-edged cylindrical steel punch (2.0 mm diameter) and a steel die with a sharp-edged aperture of small clearance (0.05 mm). Once LPS was measured, we used a 4 mm diameter punch hole to puncture disks for LMA measurements. We collected one disk per leaf (see Supplementary material for details). The disk punches dried were shipped to Tulane University, New Orleans, LA, USA to dry at 60 ℃ for 48-72 hours before being weighed.

## Molecular Work

### Tissue preservation

Upon completion of the leaf traits measurements, we prepared and preserved samples at the UC Merced Yosemite Field Station. We started by removing the main vein and margins from photosynthetic tissue. The leaf lamina was haphazardly cut with a sterile blade into 2 mm wide strip in parallel to the main vein [@arnold2003; @tellezTraits2022; @higgins2014]. Leaf strips were then sterilized with sequential washes in 95% EtOH (10 s), 0.5% sodium hypochlorite (NaOCl) (60 s), and 70% EtOH (60 s) and air dried under sterile conditions. Due to the small size of monkeyflower plants, the maximum amount of leaf lamina was preserved in sterile 15 mL tubes with \~ 10 mL CTAB solution (1 M Tris–HCl pH 8, 5 M NaCl, 0.5 M EDTA, and 20 g CTAB). The leaf tissue in CTAB solution was used for amplicon sequencing (described in detail below). All leaf tissue handling was performed in a sterile environment with an alcohol burner lamp inside a portable biosafety cabinet. All surfaces were previously sterilized sequentially with 0.5% NaOCl, 95% EtOH, and 70% EtOH. We surface sterilized surfaces and instruments in between sample handling to prevent cross contamination.

### Amplicon sequencing

We stored leaf tissue in CTAB solution for 2 months at room temperature before extracting DNA at Tulane University. To prepare for sample DNA extraction procedure, we decontaminated all instruments, materials, and surfaces in biosafety cabinet with 0.5 % NaOCl, 70 % EtOH, and 95% EtOH, and subsequently treated with UV light for 30 minutes. We subsampled 0.2 - 0.3 g of leaf tissue from each sample and placed into a sterile 2 mL tubes containing an assortment of beads: 3.2 mm stainless steel beads (Next Advance, Cat# SSB32), 100 µL stainless steel bead blend, 0.9-2.0mm (NextAdvance, Cat# SSB14B) and 2-3 of the autoclaved 2 mm zirconium oxide beads (Next Advance, Cat# ZRoB20). The 2 mL tubes with beads were previously prepared. We then proceeded to lyophilize samples for 72 hours to fully remove CTAB content from tissue. After, we submerged the sample tubes in liquid nitrogen for 30 s and homogenized samples at 30 Hz for 3 minutes in a TissueLyser LT (QIAGEN, Valencia, CA, USA). We stored samples in 20 ℃ until DNA extraction procedure.

We used a DNA extraction protocol for high-molecular weight DNA extraction adapted from Russo et al., [-@russo2022]. Briefly, it is a CTAB:chloroform:isoamyl extraction combined with a solid-phase reversible immobilization (SPRI) bead step [@russo2022; @rohland2012; @liu2023]. Protocol modifications allowed us to optimize extractions for fungal DNA from preserved leaf tissue [see details in @aponterolon2023]. After all genomic DNA was extracted, we quantified the DNA using Quant-iT® dsDNA HS Assay kit with Qubit Flourometer (Thermo Fisher Scientific, Waltham, MA, USA., Cat# Q33120) and followed a two-step amplification approach described by [-@sarmiento2017] and [-@uren2017]. We used standard primers ITS1F [@gardes1993] and ITS2 [@inbook] modified with the Illumina TruSeq adaptor. The modified primers for the first PCR (adapter ligation and ITS1 amplification) were as follows: 5’ CACTCTTTCCCTACACGACGCTCTTCCGATCTCTTGGTCATTTAGAGGAAGTAA 3’ (forward) and 5’ GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCTGCTGCGTTCTTCATCGATGC 3’ (reverse). Every sample was amplified in three parallel reactions at the annealing temperatures 52 ℃, 54 ℃, and 56 ℃ to amplify a wide range of fungal taxa and reduce amplification bias for short ITS sequences [@uren2017; @lumibao2018]. Each PCR (PCR1) reaction contained 2 µL of sample DNA template. We visualized PCR1 reactions with SYBR™ Safe DNA Gel Stain (Thermo Fisher Scientific, Waltham, MA, USA., Cat# S33102) on 2% agarose gel [@oita2021]. We combined 5 µL of amplicon product from parallel reactions in to a single tube per sample and purified using Sera-Mag™ SpeedBead Carboxylate-Modified Magnetic Particles (Hydrophobic) (Thermo Fisher Scientific, Waltham, MA, USA., Cat#09-981-123) prepared as per [-@rohland2012 and -@liu2023] and  used a ratio of 1.2x:1 with 80% EtOH following manufacturers instructions. We used 3 µL of PCR1 product from samples, DNA extraction controls, and PCR1 negative controls for a second PCR (PCR2) with barcoded adapters (IDT, Coralville, Iowa, USA). Each PCR2 reaction (total 30 µL) contained 1X Phusion Flash High Fidelity PCR Master Mix (Thermo Fisher Scientific, Waltham, MA, USA., Cat# F548L), 0.075 µM of barcoded primers (forward and reverse pooled at an initial concentration of 2 µM) and 0.20mg/mL of BSA (Thermo Fisher Scientific, Waltham, MA, USA., Cat# B14) following [-@sarmiento2017 and -@uren2017]. Before final pooling for sequencing, we purified and concentrated amplicons using SPRI beads to a total volume of 20 µL. We quantified PCR2 product with Quant-iT™ PicoGreen™ dsDNA Assay Kit (Thermo Fisher Scientific, Waltham, MA, USA., Cat# P7589) with the BioTek Synergy LX plate reader (Agilent, Santa Clara, CA) and combined equimolar amounts of libraries, including DNA extraction controls, PCR1, and PCR2 negative controls into a 10nM library pool. We did not detect any contamination visually or fluorometrically. Libraries were sequenced on the Illumina MiSeq platform with Reagent Kit v3 (2 \u00D7 300 bp) at Duke Genome Sequencing and Analysis Core Facility (Durham, NC, USA). Throughout all these steps, we used a separate set of sterile pipettes, tips, and equipment to reduce contamination in a designated PCR area to restrict contact with pre-PCR materials [@oita2021].

### Bioinformatic analyses

We assessed the quality of the reads using FastQc v0.12.1 [ v0.12.1; @andrews2010] and MultiQC [@ewels2016] tools. A total of 60,696,808 total ITS1 reads yielded from 343 (including 27 controls) libraries sequenced in two separate sequencing runs. The first sequencing event yielded 32,117,684 and the second 28,579,124 ITS1 reads. We tailored the open-source DADA2 [@callahan2016] bioinformatic pipeline for our data set. We filtered our reads for ambiguous calls before removing the adapters by using `filterAndTrim` function and argument `maxN = 0` from the `dada2` package [v1.28.0; @callahan2016]. We removed forward and reverse primer adapters (and their reverse compliments) and eliminated reads shorter than 20 bp using the `cutadapt` tool [v4.6, @martin2011]. Based on our initial quality assessment, both forward and reverse reads were of low quality, with base calls deteriorating after 100 bp. We applied stringent filter and truncation parameters to ensure quality of reads when assigning taxonomy. We filtered and truncated reads based on maximum expected errors (maxEE) rather than read length as it provides a reliable quality filtering [@edgar2015]. For this we set set the arguments `trunQ = 2`, `maxEE = c(2,20)` for forward and reverse reads, and minimum read length of 50 bp with `minLen = 50` in the used the `filterAndTrim` function [@callahan2016]. These parameters eliminated 151 samples from our data set, all from our second sequencing event. After this filter, we dereplicated reads with the `derepFastq` function and merged pairs using `mergePairs` functions with an overlap of 20 bp, minimum. We then inferred composition of the samples with `dada` function, which applies the DADA algorithm [@callahan2016; @rosen2012]. We removed chimeras via the "consensus"method with the `removeBimeraDenovo` function and ultimately we used the `assignTaxonomy` function to assign taxonomy the amplicon sequence variants (ASV) referenced against the UNITE database [@abarenkov2023a]. After taxonomy assignment we used the `phyloseq` package [@mcmurdie2013] to create a phyloseq object for downstream analyses.

We used the `decontam` package [v1.20.0; @davis2018] to statistically determine which ASVs are likely contaminants based on their frequency in our samples and remove them using `prune_taxa` function from the `phyloseq` package [v1.44.0; @mcmurdie2013]. After which, we calculated the average read count found in DNA and PCR extraction controls, considered to be laboratory contaminants, and subtracted that from the samples' read counts. We then used custom scripts to remove any ASV that represented less than 0.1% of the abundance per sample on the assumption that it originates from contamination throughout handling of samples in the DNA and PCR processes. We removed singletons ASVs with the `prune_taxa` function [@mcmurdie2013]. All these bioinformatic steps were performed in *R* [v.4.3.2; @rcoreteam2023].

## Statistical Analyses

Before performing statistical analyses, we checked for normality and homoscedasticity of the leaf traits measured. We used Shapiro-Wilk and Fligner-Killen tests from the `stats` package [@rcoreteam2023] to check for normality and homoscedasticity, respectively. We established that the data was not normally distributed and not homoscedastic. We then used non-parametric tests to compare leaf traits among species and sites. We used the Wilcoxon Rank Sum test to compare leaf trait means among species and sites. We used the `compare_means` and `'stat_compare_means`functions from the `ggpubr` package [@kassambara2023] to perform these tests and properly visualize them. We adjusted *p* values to account for false discovery rates in multiple comparisons by using “BH” method [@benjamini1995]. We performed Principal Component Analysis (PCA) to understand patterns and relationships among leaf traits of host species. We used the `prcomp` function from the `vegan` package [@oksanen2022] compute the PCA analysis with variables ACI, LT, LPS, LMA, and LBI, all log-transformed. We used only complete raw leaf functional traits measurements to compute the PCA analysis (*n* = 504), samples with missing values were omitted. 

To facilitate our understanding of the effects of elevation on FEF communities, we categorized elevation as "low" (< 1219 m.a.s.l), "medium" (1220 - 1828 m.a.s.l.) and "high" (> 1829 m.a.s.l.).  We then calculated alpha diversity per sample as the observed richness and as Shannon diversity index with `phyloseq`'s `estimated-richness` function [@mcmurdie2013]. We used the `vegdist` function from the `vegan` package [@oksanen2022] to calculate the Bray-Curtis dissimilarity matrix, a measure of beta diversity, to compare the FEF communities within each monkeyflower species across sites and elevation. Additionally, we used a distance-based Redundancy Analysis (dbRDA) to statistically compare the FEF community similarities within each species per site. We modeled the dbRDA analysis with all leaf functional traits included and elevation as a continuous variable. Its visualizations effectively portray underlying patterns of compositional differences [@anderson2017;@legendre1999; @mcardle2001], akin to permutational analysis of variance. We used the `anova.cca` function to assess the marginal significance of constraining variables [@legendre2011; @legendre2012; @oksanen2022). We corroborated homogeneous dispersion of variances with a permutational analysis of multivariate dispersion (PERMDISP) using the `betadisper` and `permutest` functions from the `vegan` package [@oksanen2022]. We used the `simper` function from the `vegan` package to discriminate which species contribute the most to compositional differences between groups [@oksanen2022].

Ultimately, we calculated distance matrix of geographical distances between sites using the Haversine method with the `geosphere` package [@hijmans2022]. To test the correlation between the geographical distance matrix and FEF community dissimilarity (Bray-Curtis dissimilarity) `mantel` function from the `vegan` package [@oksanen2022]. All statistical analyses were performed in *R* programming language[v.4.3.2; @rcoreteam2023].

# Results

We obtained 5,082,229 reads representing 726 ASVs from 174 samples after processing samples through the DADA2 pipeline. The raw reads obtained were composed of 26.81% Ascomycota, 71.53% Basidiomycota, <0.05% Chytridiomycota,  <0.5% Mortierellomycota, <0.03% Olpidiomycota, 0.01% Rozellomycota, <0.001% Aphelidiomycota, and 1.19% missed BLAST hits. After decontamination, and removal of singletons we obtained 4,971,645 reads representing 231 ASVs from 160 samples composed of 25.88% Ascomycota, 73.1% Basidiomycota, 0.01% Chytridiomycota,  <0.1% Mortierellomycota, 0.02% Olpidiomycota, <0.002% Rozellomycota and <1.0% unknown reads. 

NEEDS TO BE UPDATED.


# Discussion

# Conclusion

# Author Contributions

# Acknowledgements

# Conflict of Interest Statement

The authors declare no competing interests.

# Data Availability Statement

# References

::: {#refs}
:::

# Figures

```{r, leaf_traits}
#| echo: false
#| eval: true
#| tidy: true

library("tidyverse")
library("data.table")
library("ggplot2")
library("ggpubr")
library("ggfortify")
library("ggthemes")
library("ggtext")
library("ggiraphExtra")
library("MetBrewer")

library("rstatix")
library("car")
library("nlme")

library("vegan")
library("phyloseq")
library("hillR")
library("metagMisc")
library("iNEXT")
library("geosphere")


path <-"/home/baponte/Boxx/Dissertation/Mimulus/Data/CH2/bioinformatics"
traits <- read.csv(file.path(path, "mim2_leaf_traits_field_survey.csv"))

#Elevation_m categories (meters)
breaks <- c(-Inf, 1219.2, 1828.495, Inf)
labels <- c("LOW", "MEDIUM", "HIGH")

#Leaf level data set
leaf_traits <-  traits |>
  slice(-c(959:973))|> #Removing control samples
  rename(LMA = LMA.mg., 
         LBI = Leaf_lobe_index,
         LPS = Leaf_toughness,
         LT = Leaf_thickness) |> 
  select(!c(5,12:15, 23:25)) |>
  group_by(Site, Species, Unique_ID) |>
  fill(ImageJ_1, ImageJ_2, LBI, .direction = 'down') |>
  mutate(Species = as.factor(Species), 
         Site = as.factor(Site),
         Elevation_cat = cut(Elevation_m, 
                             breaks = breaks,
                             labels = labels)) |>
  relocate(Elevation_cat, .before = Elevation_m)|>
  as_tibble()

# Save file
#saveRDS(leaf_traits, file.path(path, "statistics/leaf_traits.rds"))
#write.csv(leaf_traits, file.path(path, "statistics/leaf_traits.csv"))

# Load file
leaf_traits <- readRDS(file.path(path, "statistics/leaf_traits.rds"))


#Plant level for ASV analyses
plant_traits <- leaf_traits |>
  relocate(Unique_ID, .before = Site) |>
  group_by(Unique_ID, Site, Species, Elevation_m, Elevation_cat, Longitude, Latitude)|>
  summarise(ACI = mean(ACI, na.rm = TRUE),
            Thickness = mean(LT, na.rm = TRUE),
            Toughness = mean(LPS, na.rm =TRUE),
            LMA = mean(LMA, na.rm = TRUE),
            ImageJ_1 = mean(ImageJ_1, na.rm = TRUE),
            ImageJ_2 = mean(ImageJ_2, na.rm = TRUE),
            LBI = mean(LBI, na.rm = TRUE))

# Save file
#saveRDS(plant_traits, file.path(path, "statistics/plant_traits.rds"))
#write.csv(plant_traits, file.path(path, "statistics/plant_traits.csv"))

# Load file
plant_traits <- readRDS(file.path(path, "statistics/plant_traits.rds"))


#getting rownmes to filter
#names <- column_to_rownames(all_decontaminated, var = "X")
#samples <- rownames(names) #Using the colname to filter out 

# Leaf traits transformed
leaf_traits <- leaf_traits |>
  mutate(logLBI = log10(LBI),
         logACI = log10(ACI),
         logLT = log10(LT),
         logLPS = log10(LPS),
         logLMA = log10(LMA))

#Removing species B
leaf_traits_noB <- leaf_traits |>
  filter(!Species == "B")

# Plant traits transformed
plant_traits <- plant_traits |>
  mutate(logLBI = log10(LBI),
         logACI = log10(ACI),
         logLT = log10(LT),
         logLPS = log10(LPS),
         logLMA = log10(LMA))

## Phyloseq objects

#Phyloseq joins various objects that we have already prepare: taxonomic table, ASV table and our sample data.
#These are the data frames resulting from the phyloseq section in mim2_bioinformatics.qmd notebook

#Phyloseq object cleaned of singletons (231 ASVs)
ps_clean_3 <- readRDS(file.path(path, "taxonomy/ASV_8450_assigned_nonsingletons.rds"))
ps_clean_3_df  <- read.csv(file.path(path, "taxonomy/ASV_8450_assigned_nonsingletons.csv"))

#Relative abundance phyloseq
rel_abund_ps_clean <- readRDS(file.path(path, "taxonomy/ASV_8450_relabun_nonsingletons.rds"))
rel_ps_clean_df <- read.csv(file.path(path, "taxonomy/ASV_8450_relabun_nonsingletons.csv"))

# Leaf traits with community data ####
plant_traits <- ps_clean_3 |> #Estimating community richness and shannon diversity from phyloseq object
  estimate_richness(split = TRUE, measures = c("Observed","Shannon")) |>
  rownames_to_column() |>
  rename(Unique_ID = rowname) |>
  right_join(plant_traits, by = "Unique_ID") |>
  relocate(Observed, Shannon, .after = logLMA)
 
# Matrices
# ASV abundance matrix
asv_matrix <- ps_clean_3_df |>
  select(!c(2:8)) |>
  column_to_rownames(var = "OTU") |>
  as.matrix()

names_list <- colnames(asv_matrix)

#Bray-Curtis dissimilarity matrix
asv_dist <- vegdist(t(asv_matrix), method = "bray", binary = FALSE, upper = FALSE) |>
  as.matrix()

# Adding column and rownames
rownames(asv_dist) <- names_list
colnames(asv_dist) <- names_list

# Geographical matrix
#geo_matrix <- distm(plant_traits[,c("Longitude", "Latitude")], fun = distHaversine)
geo_matrix <- distm(plant_traits |>
                      filter(Unique_ID %in% names_list) |>
                     # column_to_rownames(var = "Unique_ID") |>
                      select(Longitude, Latitude), fun = distHaversine)
                      
# Adding column and rownames
rownames(geo_matrix) <- names_list
colnames(geo_matrix) <- names_list

# Converting to distance object
#geo_dist <- as.dist(geo_matrix)



#Subset data set for NMDS plot
nmds_traits <- plant_traits |>
  filter(Unique_ID %in% names_list)
```


## Figure 1

```{r, figure1}
#| echo: false
#| eval: true
#| tidy: true
#| fig-width: 6
#| fig-height: 6
#| label: figure1
#| fig-cap: "PCA of leaf traits by species."

# PCA using covariates to explain species richness/abundance ####
data.pca <- leaf_traits_noB |>
  na.omit() |>
  select(c(19:23))

leaf_traits.pca <- na.omit(leaf_traits_noB) 


###Run this to create pca with prcomp function
pca <- prcomp(data.pca, scale = TRUE)
pca$rotation=-pca$rotation
pca$x=-pca$x

#Checking the PCA
#plot(pca,type = "lines")
#biplot(pca) # Base type PCA

# PCA using autoplot() and prcomp()and modifying with ggplot syntax ####
pca_auto <- autoplot((pca), data = data.frame(leaf_traits.pca), # For some reason this functions require data to be data= data.frame()
         alpha = 0, #Setting alpha to zero render the automatic circle point null.Manipulate shapes with geom_point(). 
         loadings = TRUE,loadings.colour = "black",
         loadings.label = TRUE, loadings.label.colour="black",
         loadings.label.size = 4, size = 7, 
         loadings.label.vjust = 0, 
         loadings.label.hjust = -0.15) + 
  geom_point(aes(fill = Species, color = Species), alpha = 0.5, size = 3) +
  geom_hline(yintercept = 0, colour = "gray45") +
  geom_vline(xintercept = 0, colour = "gray45") +
  scale_fill_manual(labels=c('*M. guttatus*', '*M. laciniatus*', '*M. nasutus*'), 
                    values = met.brewer("Tam", n = 4, type = "discrete", direction = c(-1))) +
  scale_color_manual(labels = c('*M. guttatus*', '*M. laciniatus*', '*M. nasutus*'),
                     values = met.brewer("Tam", n = 4, type = "discrete", direction = c(-1))) +
  stat_ellipse(aes(color = Species), 
               geom = "path", 
               size = 1.3, 
               position = "identity", 
               type = "t", 
               linetype = 1, 
               level = 0.95, 
               segments = 51, na.rm = TRUE, show.legend = NA, inherit.aes = TRUE) +
  #scale_x_continuous(expand = c(0.0, -0.09)) +
  theme_classic(base_size = 12) +
  theme(legend.text = element_markdown(),
        legend.position = "bottom",
        legend.title = element_text(face = "bold")) 
  # labs(caption = expression("ACI = anthocyanins, LPS = leaf punch strength, LMA = leaf mass per area, LT = leaf thickness. LBI = Leaf lobe index. All leaf replicates per species ("~ italic("n") ~ "= 501)."))


pca_auto$layers <- c(pca_auto$layers, pca_auto$layers[[2]], pca_auto$layers[[3]]) # This adds/copies layers 2-3 and overlays them. It makes the arrows be on top of the points. There must be a better ways of doing this.

pca_auto

#ggsave(filename = file.path(path, "figures/pca.png"), plot = pca_auto, dpi=300, units=c("mm"), width=150, height=150)
```

## Figure 2

```{r, figure2}
#| echo: false
#| eval: true
#| tidy: true
#| fig-width: 6
#| fig-height: 6
#| label: figure2
#| fig-cap: "Scatter plots of Shannon Diversity Index by elevation category."


# Formatting formula for p value
format.p <- function(p, precision = 0.001) {
  digits <- -log(precision, base = 10)
  p <- formatC(p, format = 'f', digits = digits)
  if (p < 0.001) {
    p = paste0('< ', precision)}
  if (p >= 0.001) {
    p = paste0('= ', p)    }
  sub("0", "0", p)
}


#Legend title
legend_title <- "Species"

# Shannon diversity

# Linear model summary
#summary(lm(Shannon ~ Elevation_m, data = plant_traits))
elev_div_p <- format.p(cor.test(plant_traits$Shannon, plant_traits$Elevation_m)$p.value) # This is just to make the p-value nicer. 


elev_diversity <- ggplot(data = plant_traits  |>
  filter(!Species == "B"), 
  aes(x = Elevation_m, y = Shannon, color = Species)) +
   geom_jitter(aes(color = plant_traits$Species[plant_traits$Species != "B"]), size = 3, alpha = 0.7) + 
   geom_smooth(method = lm, se = F) + 
   # stat_regline_equation(aes(label = paste(..eq.label.., ..adj.rr.label.., sep = '~~~~'))) 
   geom_smooth(method = lm, se = T, level = 0.95, na.rm = F, color = "black") + 
   stat_regline_equation(label.y = 3, label.x = 1000, 
                         aes(label = paste(..eq.label.., sprintf("italic('p')~'%s'", elev_div_p), 
                                          sep = "~~~~")), size = rel(4), color = "black") +
   stat_regline_equation(label.y = 2.7, label.x = 1900, 
                          aes(label = ..adj.rr.label.., ), size = rel(4), color = "black") + 
   scale_fill_manual(labels = c("*M. laciniatus*","*M. nasutus*", "*M. guttatus*"), 
                     values = met.brewer("Tam", n = 4, type = "discrete", direction = c(-1))) + 
   scale_color_manual(legend_title, labels = c("*M. laciniatus*", "*M. nasutus*", "*M. guttatus*"), 
                      values = met.brewer("Tam", n = 4, type = "discrete", direction = c(-1))) + 
   labs(y = "Shannon diversity", x = "Elevation (m.a.s.l.)", col = "") + 
   theme_classic(base_size = 12) +
   theme(legend.title = element_text(face = "bold"),
         legend.position = "bottom",
         panel.border = element_rect(linetype = "blank", fill = NA),
         legend.text = element_markdown(size = 12), #Make legend text italic
         axis.text.x = element_text(size = 12),
         axis.title.x = element_text(size = 14))

elev_diversity

#ggsave(filename = file.path(path, "figures/shannon_elevation.png"), plot = elev_diversity, dpi=300, units=c("mm"), width=250, height=150)
```

## Figure 3

```{r, figure3}
#| echo: false
#| eval: true
#| tidy: true
#| fig-width: 6
#| fig-height: 6
#| label: figure3
#| fig-cap: "Scatter plot of Shannon diversity by elevation."

#Linear model summary
#summary(lm(Observed ~ Elevation_m, data = plant_traits))
richness_p <- format.p(cor.test(plant_traits$Observed, plant_traits$Elevation_m)$p.value) # This is just to make the p-value nicer. 


rich_diversity <- ggplot(data = plant_traits  |>
  filter(!Species == "B"), 
  aes(x = Elevation_m, y = Observed, color = Species)) +
   geom_jitter(aes(color = plant_traits$Species[plant_traits$Species != "B"]), size = 3, alpha = 0.7) + 
   geom_smooth(method = lm, se = F) + 
   # stat_regline_equation(aes(label = paste(..eq.label.., ..adj.rr.label.., sep = '~~~~'))) 
   geom_smooth(method = lm, se = T, level = 0.95, na.rm = F, color = "black") + 
   stat_regline_equation(label.y = 50, label.x = 1000, 
                         aes(label = paste(..eq.label.., sprintf("italic('p')~'%s'", richness_p), 
                                          sep = "~~~~")), size = rel(4), color = "black") +
  stat_regline_equation(label.y = 45, label.x = 1900, 
                          aes(label = ..adj.rr.label..), size = rel(4), color = "black") + 
   scale_fill_manual(labels = c("*M. laciniatus*","*M. nasutus*", "*M. guttatus*"), 
                     values = met.brewer("Archambault", n = 4, type = "discrete", direction = c(1))) + 
   scale_color_manual(legend_title, labels = c("*M. laciniatus*", "*M. nasutus*", "*M. guttatus*"), 
                      values = met.brewer("Tam", n = 4, type = "discrete", direction = c(-1))) + 
   labs(y = "Observed Richness", x = "Elevation (m.a.s.l.)", col = "") + 
   theme_classic(base_size = 12) +
   theme(legend.title = element_text(face = "bold"),
         legend.position = "bottom",
         panel.border = element_rect(linetype = "blank", fill = NA),
         legend.text = element_markdown(size = 12), #Make legend text italic
         axis.text.x = element_text(size = 12),
         axis.title.x = element_text(size = 14)) 
  # annotate(geom = "text", x = 2100, y = 2.5, label = "p = 0.03", color = "black")
#annotate(geom="text", x=2100, y=50, label="p = 0.06", color = "black")


rich_diversity


#ggsave(filename = file.path(path, "figures/richness_elevation.png"), plot = rich_diversity, dpi=300, units=c("mm"), width=250, height=150)
```


## Figure 4

```{r, figure4}
#| echo: false
#| eval: true
#| tidy: true
#| fig-width: 6
#| fig-height: 6
#| label: figure4
#| fig-cap: "Bar plot of relative abundance of fungal phyla by species and elevation category."

ps_melt <- psmelt(ps_clean_3) # Melting data to create accurate relative abunddance plot
p1 <- ggplot(ps_melt |>
  filter(!sample_Species == "B"), aes(x = sample_Species, y = Abundance)) +
  geom_bar(aes(fill = Phylum), stat="identity", position = "fill") +
  scale_fill_manual(labels=c('Ascomycota', 'Basidiomycota', 'Chytridiomycota', 'Fungi_ord_Incertae_sedis', 'Mortierellomycota', 'Olpidiomycota'),values= met.brewer("Lakota", n = , type = "discrete", override.order = TRUE)) +
  theme_classic(base_size = 14) +
  theme(strip.background = element_rect(fill = "white"),
        strip.text.x = element_text(colour = "black", face = "bold"),
        legend.text = element_text(size = 12),
        panel.border = element_rect(linetype = "blank", fill = NA),
        axis.text.y = element_text(size = 12),
        axis.text.x = element_text(face="italic", 
                                    size = 10,  
                                    hjust = 1,
                                    angle = 30)) +
  scale_x_discrete(labels = c('M. laciniatus', 'M. nasutus', 'M. guttatus')) +
  labs(y = "Abundance", x = "", caption = "n = 160") +
  facet_grid(~Elevation_cat)

  
p1 
#ggsave(file.path(path, "figures/relabun_barplot.png") , plot = p1, dpi=300, units=c("mm"), width=200, height=150)
```


## Figure 5
```{r, figure5}
#| echo: false
#| eval: true
#| tidy: true
#| warning: false
#| fig-width: 6
#| fig-height: 6
#| label: figure5
#| fig-cap: "Violin plot of Shannon diversity by species."

#Comparisons
my_comparisons <- list(c("N", "L"), c("N", "G"), c("L", "G"))

#Richness 
pp <- ggplot(plant_traits |>
               filter(!Species == "B"), aes(x = Species, y = Shannon)) +
  geom_violin(aes(fill = Species)) +
  #geom_boxplot(width = 0.3, color = "grey30", alpha = 0.2) +
  geom_crossbar(stat = "summary",
                fun = mean,
                fun.max = mean,
                fun.min = mean,
                fatten = 1.5, width = 0.8) +
  geom_point(color = "black", alpha = 0.2,
             size = 1.7,
             position = position_jitter(w = 0.05))  +
  scale_fill_manual(labels =c('*M. laciniatus*', '*M. nasutus*', '*M. guttatus*'),
                    values = met.brewer("Tam", n = 4, type = "discrete", direction = c(-1))) +
  theme_classic(base_size = 12) +
  theme(legend.title = element_text(face = "bold"),
        panel.border = element_rect(linetype = "blank", fill = NA),
        legend.text = element_markdown(size = 12)) +
  scale_x_discrete(labels = c('', '', '', '')) +
  stat_compare_means(comparisons = my_comparisons,
                     method = "wilcox.test", 
                     label = "p.signif") +
  labs(y = "Shannon Diversity", x = "", caption = "n = 160") +
  facet_wrap(~Elevation_cat)
pp

#ggsave(file.path(path, "figures/shannon_diversity.png") , plot = pp, dpi=300, units=c("mm"), width=200, height=150)
```

## Figure 6

```{r, figure6}
#| echo: false
#| eval: true
#| tidy: true
#| fig-width: 6
#| fig-height: 6
#| label: figure6
#| fig-cap: "FEF community composition association with leaf functional traits, especially log(LBI) and Elevation, per host species. Distance-based Redundancy Analysis (dbRDA) plot of leaf functional traits by species. Each point represents a FEF community sample from one host species; colors represent host species"

m2dbRDA <- dbrda(asv_dist ~ logACI + logLT + logLPS + logLMA + logLBI + Elevation_m, distance = "bray", data = nmds_traits, na.action = na.omit)

#summary(m2dbRDA)
#plot(m2dbRDA)
#anova(m2dbRDA, by = "margin") 
#anova(m2dbRDA, by = "axis")
#anova(m2.forsel, by="axis", perm.max=500)

#names(m2dbRDA)
B <- summary(m2dbRDA)

# Extracting scores for the plot
A.1 <- vegan::scores(m2dbRDA)
A.2 <- as.data.frame(A.1$sites) #converting to df

# Extracting names for the plot
dbrda_names <- rownames(A.2)

# Row names to column
A.2 <- rownames_to_column(A.2, var = "Unique_ID")

# Subset for the plot
dbrda_df <- nmds_traits |>
  filter(Unique_ID %in% dbrda_names) 

A.3 <- right_join(A.2, dbrda_df, by = "Unique_ID") 
A.3 <- A.3 |>
  filter(!Species == "B")

# Scores for arrows
A.4 <- data.frame(vegan::scores(m2dbRDA, display = "bp"))


# Define the arrow aesthetic mapping
arrow_map <- aes(xend = (dbRDA1*2), yend = (dbRDA2*2), x = 0, y = 0, shape = NULL, color = NULL, fill = NULL)
label_map <- aes(x = 2*dbRDA1, y = 2*dbRDA2, label = row.names(A.4), shape = NULL, color = NULL, fill = NULL)
arrowhead = arrow(length = unit(0.02, "npc"))

#subset A4 for labeling
A.4 <- A.4[sort(rownames(A.4)),]

A4.sub1 <- A.4[1,] #Elevation
A4.sub2 <- A.4[2,] #Anthocyanins
A4.sub3 <- A.4[3,] #LBI
A4.sub4 <- A.4[4,] #LMA
A4.sub5 <- A.4[5,] #LPS
A4.sub6 <- A.4[6,] #LT.


#A4.sub4

# dbRDA plot based on species
p_m2dbrda <- ggplot(data = A.3, aes(x = dbRDA1, y = dbRDA2)) +
  geom_point(data = A.3, aes(color = Species, shape = Species), size = rel(4), alpha = 0.7, stroke = 0.5) +
  geom_segment(arrow_map, size = rel(.8), data = A.4, color = "black", arrow = arrowhead) +
  geom_vline(xintercept = 0, color = "grey30", linetype = "dashed") + # plot vertical line
  geom_hline(yintercept = 0, color = "grey30", linetype = "dashed") + # plot horizontal line
  geom_text(label_map, size = rel(4), data = A4.sub1, fontface = "bold", show.legend = FALSE, label = c("Elv."), nudge_x = -0.1) +
  geom_text(label_map, size = rel(4), data = A4.sub2, fontface = "bold", show.legend = FALSE, label = c("logACI"), nudge_y = 0.1, nudge_x = -0.1) +
  geom_text(label_map, size = rel(4), data = A4.sub3, fontface = "bold", show.legend = FALSE, label = c("logLBI"), nudge_y = 0.1) +
  geom_text(label_map, size = rel(4), data = A4.sub4, fontface = "bold", show.legend = FALSE, label = c("logLMA"), nudge_y = 0.1, nudge_x = -0.1) +
  geom_text(label_map, size = rel(4), data = A4.sub5, fontface = "bold", show.legend = FALSE, label = c("logLPS"), nudge_y = 0.1, nudge_x = 0.2) +
  geom_text(label_map, size = rel(4), data = A4.sub6, fontface = "bold", show.legend = FALSE, label = c("logLT"), nudge_x = 0.2) +
  xlab(label = paste("dbRDA1 (", round(B$concont$importance[2,1]*100, digits = 1), "%)", sep="")) +
  ylab(label = paste("dbRDA2 (", round(B$concont$importance[2,2]*100, digits = 1), "%)", sep="")) + 
  scale_color_manual(labels=c('*M. laciniatus*', '*M. nasutus*', '*M. guttatus*'),
                    values= met.brewer("Tam", n = 3 , type = "discrete", direction = -1)) +
  scale_shape_manual(values=c(15:18),labels=c('*M. laciniatus*', '*M. nasutus*', '*M. guttatus*')) +
  stat_ellipse(aes(color = Species), 
               geom = "path", 
               size = 1.7, 
               position = "identity",
               type = "t", linetype = 1,
               level = 0.95, 
               segments = 51,
               na.rm = FALSE, 
               show.legend = NA, inherit.aes = TRUE) +
  theme_classic(base_size = 12) +
  theme(legend.title = element_text(face = "bold"),
        legend.position = "bottom",
        legend.text = element_markdown()) 
  #guides(color = FALSE, size = FALSE)
  #labs(caption = "ACI = anthocyanins, LPS = leaf punch strength, LMA = leaf mass per area, LT = leaf thickness, LBI = leaf lobe index, Elv. = Elevation_m (f.a.s.l.)", hjust = 0)
     
# Rearranging the order of layers so arrows are above the points.

p_m2dbrda$layers <- c(p_m2dbrda$layers, p_m2dbrda$layers[[2]], p_m2dbrda$layers[[5]], p_m2dbrda$layers[[6]], p_m2dbrda$layers[[7]], p_m2dbrda$layers[[8]])
#p_m2dbrda

#ggsave(filename=file.path(path, "figures/dbRDA_species.png"), plot = p_m2dbrda, dpi=600, units=c("mm"), width=150, height=150)


# dbRDA plot based on elevation
 elev_dbrda <- ggplot(data = A.3, aes(x = dbRDA1, y = dbRDA2)) +
  geom_point(data = A.3, aes(color = Elevation_cat, shape = Species), size = rel(4), alpha = 0.7, stroke = 0.5) +
  geom_segment(arrow_map, size = rel(.8), data = A.4, color = "black", arrow = arrowhead) +
  geom_vline(xintercept = 0, color = "grey30", linetype = "dashed") + # plot vertical line
  geom_hline(yintercept = 0, color = "grey30", linetype = "dashed") + # plot horizontal line
  geom_text(label_map, size = rel(4), data = A4.sub1, fontface = "bold", show.legend = FALSE, label = c("Elv."), nudge_x = -0.1) +
  geom_text(label_map, size = rel(4), data = A4.sub2, fontface = "bold", show.legend = FALSE, label = c("logACI"), nudge_y = 0.1, nudge_x = -0.1) +
  geom_text(label_map, size = rel(4), data = A4.sub3, fontface = "bold", show.legend = FALSE, label = c("logLBI"), nudge_y = 0.1) +
  geom_text(label_map, size = rel(4), data = A4.sub4, fontface = "bold", show.legend = FALSE, label = c("logLMA"), nudge_y = 0.1, nudge_x = -0.1) +
  geom_text(label_map, size = rel(4), data = A4.sub5, fontface = "bold", show.legend = FALSE, label = c("logLPS"), nudge_y = 0.1, nudge_x = 0.2) +
  geom_text(label_map, size = rel(4), data = A4.sub6, fontface = "bold", show.legend = FALSE, label = c("logLT"), nudge_x = 0.2) +
  xlab(label = paste("dbRDA1 (", round(B$concont$importance[2,1]*100, digits = 1), "%)", sep="")) +
  ylab(label = paste("dbRDA2 (", round(B$concont$importance[2,2]*100, digits = 1), "%)", sep="")) + 
  scale_color_manual(labels=c('LOW', 'MEDIUM', 'HIGH*'),
                    values= met.brewer("Lakota", n = 3 , type = "discrete", direction = -1)) +
  scale_shape_manual(values=c(15:18),labels=c('*M. laciniatus*', '*M. nasutus*', '*M. guttatus*')) +
  stat_ellipse(aes(color = Elevation_cat), 
               geom = "path", 
               size = 1.7, 
               position = "identity",
               type = "t", linetype = 1,
               level = 0.95, 
               segments = 51,
               na.rm = FALSE, 
               show.legend = NA, inherit.aes = TRUE) +
  theme_classic(base_size = 12) +
  theme(legend.title = element_text(face = "bold"),
        legend.position = "bottom",
        legend.text = element_markdown()) 
  #guides(color = FALSE, size = FALSE)
  #labs(caption = "ACI = anthocyanins, LPS = leaf punch strength, LMA = leaf mass per area, LT = leaf thickness, LBI = leaf lobe index, Elv. = Elevation_m (f.a.s.l.)", hjust = 0)
     
# Rearranging the order of layers so arrows are above the points.

elev_dbrda$layers <- c(elev_dbrda$layers, elev_dbrda$layers[[2]], elev_dbrda$layers[[5]], elev_dbrda$layers[[6]], elev_dbrda$layers[[7]], elev_dbrda$layers[[8]])
#elev_dbrda

double_dbrda <- ggarrange(p_m2dbrda, elev_dbrda, ncol = 1, nrow = 2, 
          labels = c("A", "B"),
          common.legend = F, legend = "right")

double_dbrda
#Save the plot
#ggsave(file.path(path, "figures/dbRDA_species_elevation.png"), plot = double_dbrda, dpi=300, units=c("mm"), width=180, height=200)
```


## Figure 7
```{r, figure7}
#| echo: false
#| eval: true
#| tidy: true
#| fig-width: 6
#| fig-height: 6
#| label: figure7
#| 
my_comparisons3 <- list(c("LOW", "MEDIUM"), c("LOW", "HIGH"), c("MEDIUM", "HIGH"))

bbe <-  ggplot(plant_traits |>
                filter(!Species == "B"), aes(x = Elevation_cat, y = Shannon)) +
  
  geom_violin(aes(fill = Elevation_cat)) +
  geom_crossbar(stat="summary", fun = mean, fun.max = mean, fun.min = mean,  fatten = 2, width = 0.5, color="black") +
  geom_point(aes(shape = Species),color="black", alpha = 0.4,size=1.7, position = position_jitter(w=0.05)) +
  scale_color_manual(labels=c('Low', 'Medium', 'High'),values= met.brewer("Lakota", n = 3, type = "discrete")) +
  scale_fill_manual(labels=c('Low', 'Medium', 'High'),values= met.brewer("Lakota", n = 3, type = "discrete")) +
  scale_shape_manual(values=c(15:18),labels=c('*M. laciniatus*', '*M. nasutus*', '*M. guttatus*')) +
  theme_classic(base_size = 14) +
  theme(strip.text.x = element_text(colour = "black", face = "bold"),
        legend.title = element_text(face = "bold"),
        panel.border = element_rect(linetype = "blank", fill = NA),
        legend.text = element_markdown(),
        axis.text.x = element_text(face="italic", size = 14)) +
  scale_x_discrete(labels = c('', '', '', '')) +
  stat_compare_means(comparisons = my_comparisons3,
                     method = "wilcox.test", 
                     label = "p.signif") +
  labs(y = "Shannon Diversity", x = "", caption = "n = 160")
bbe

#ggsave(file.path(path, "figures/shannon_elevation.png") , plot = bbe, dpi=300, units=c("mm"), width=200, height=150)
```


# Figure 8

```{r, figure8}
#| echo: false
#| eval: true
#| tidy: true
#| fig-width: 6
#| fig-height: 6
#| label: figure8
#| fig-cap: "Scatter plot of Bray-Curtis dissimilarity by geographic distance."

# Prepare the data for correlation plot
# Longer format
asv_bray <- asv_dist |>
  as_tibble(rownames = "A") |>
  pivot_longer(-A, names_to = "B", values_to = "bray_diss")

geo_long <- geo_matrix |>
  as_tibble(rownames = "A") |>
  pivot_longer(-A, names_to = "B", values_to = "geo_dist")

# Merge the two dataframes
asv_geo_matrix_long <- inner_join(asv_bray, geo_long, by = c("A", "B")) |>
  inner_join(plant_traits |>
               dplyr::select(Unique_ID, Site, Species, Elevation_cat), by = c("A" = "Unique_ID"))

# Visualize the correlation
mantel_test <- ggplot(asv_geo_matrix_long |>
         filter(A < B, !Species == "B"), aes(y = bray_diss, x = geo_dist)) +
  geom_jitter(aes(color = Species), alpha = 0.5, size = 3) +
  geom_smooth(method = "lm", se = F, color = "black") +
   #stat_regline_equation(label.y = 0.5, label.x = 1900, 
                          #aes(label = ..eq.label..), size = rel(4), color = "black") +
  #scale_fill_manual(labels = c("*M. laciniatus*","*M. nasutus*", "*M. guttatus*"), 
                     #values = met.brewer("Archambault", n = 4, type = "discrete", direction = c(1))) + 
   scale_color_manual("Species", labels = c("*M. laciniatus*", "*M. nasutus*", "*M. guttatus*"), 
                      values = met.brewer("Tam", n = 4, type = "discrete", direction = c(-1))) + 
   labs(y = "Bray_Curtis dissimilarity", x = "Geographic distance", col = "") + 
   theme_classic(base_size = 12) +
   theme(legend.title = element_text(face = "bold"),
        panel.border = element_rect(linetype = "blank", fill = NA),
        legend.text = element_markdown(size = 12), #Make legend text italic
        axis.text.x = element_text(size = 12),
        axis.title.x = element_text(size = 14))

mantel_test

ggsave(file.path(path, "figures/mantel_test.png") , plot = mantel_test, dpi=300, units=c("mm"), width=200, height=150)

# Mantel test
abund_geo  = mantel(asv_dist, geo_matrix, method = "spearman", permutations = 9999, na.rm = TRUE)
#abund_geo
```


# Supplementary Material


## Figure S1

```{r, supp_figS1}
#| echo: false
#| eval: true
#| tidy: true
#| fig-width: 6
#| fig-height: 6
#| label: supp_figS1
#| fig-cap: "Violin plots of leaf mass per area (LMA) by species and elevation category."

#ggpubr package used for comparisons tests
lma_compare_means <- compare_means(logLMA ~ Species,
         data = leaf_traits_noB,
         group.by = "Elevation_cat",
         method = "wilcox.test",
         p.adjust.method = "BH",
  paired = FALSE,
  alternative = "two.sided", 
  mu = 0, 
  var.equal = FALSE,
  conf.level = 0.95)


#lma_compare_means

#Report with `compare_means` because it is easier to plot with `ggpubr` package.

#Error bars for plots 
errbar_lims <- group_by(leaf_traits_noB, Site) |> 
              summarize(mean = mean(logLMA), se = sd(logLMA)/sqrt(n()), 
                        upper = mean+(2*se), lower = mean-(2*se))
#Comparisons
my_comparisons <- list(c("N", "L"), c("N", "G"), c("L", "G"))

#By Site
lma <- ggplot(leaf_traits_noB, aes(y = logLMA, x = Species)) +
  geom_violin(aes(fill = Species)) +
  geom_crossbar(stat = "summary",
                fun = mean,
                fun.max = mean,
                fun.min = mean,
                fatten = 1.5, width = 0.8) +
  geom_point(color = "black", alpha = 0.2,
             size = 1.7, 
             position = position_jitter(w = 0.05))  +
  scale_fill_manual(labels =c('*M. laciniatus*', '*M. nasutus*', '*M. guttatus*'),
                    values = met.brewer("Tam", n = 4, type = "discrete", direction = c(-1))) +
  scale_color_manual(aes(color = "black")) +
  #scale_color_manual(labels=c(''*M. laciniatus*', '*M. nasutus*', '*M. guttatus*'),values= met.brewer("Archambault", n = 4, type = "discrete")) +
  theme_classic(base_size = 12) +
  theme(legend.title = element_text(face = "bold"),
        panel.border = element_rect(linetype = "blank", fill = NA),
        legend.text = element_markdown(size = 12), #Make legend text italic
        # axis.text.x = element_text(face="italic", 
        #                             size = 10,  
        #                             hjust = 1,
        #                             angle = 30),
        axis.text.x = element_blank()) +
  labs(y = "log10[Leaf Mass per Area (LMA) (mg/mm)]", x = "") +
  #scale_x_discrete(labels = c('M. laciniatus', 'M. nasutus', 'M. guttatus')) +
  stat_compare_means(comparisons = my_comparisons,
                     method = "wilcox.test",
                     #label.y = c(0.4, 0.6, 1),
                     label = "p.signif") +
  facet_wrap(~Elevation_cat)

lma

#ggsave(filename = file.path(path, "figures/lma_violin.png"), plot = lma, dpi=300, units=c("mm"), width=300, height=150)
```

# Figure S2

```{r, supp_figS2}
#| echo: false
#| eval: true
#| tidy: true
#| fig-width: 6
#| fig-height: 6
#| label: supp_figS2
#| fig-cap: "Violin plots of Anthocyanin Content Index (ACI) by species and elevation category."

#ggpubr package used for comparisons tests
aci_compare_means <- compare_means(logACI ~ Species,
         data = leaf_traits_noB,
         group.by = "Elevation_cat",
         method = "wilcox.test",
         p.adjust.method = "BH",
  paired = FALSE,
  alternative = "two.sided", 
  mu = 0, 
  var.equal = FALSE,
  conf.level = 0.95)

#aci_compare_means
#Roughly the same outcome as pairwise_t_test.

#Report with `compare_means` because it is easier to plot with `ggpubr` package.
errbar_lims <- group_by(leaf_traits_noB, Species) |> 
              summarize(mean = mean(logACI), se = sd(logACI)/sqrt(n()), 
                        upper = mean + (2*se), lower = mean - (2*se))

#Comparisons
my_comparisons <- list(c("N", "L"), c("N", "G"), c("L", "G"))


aci <- ggplot(leaf_traits_noB, aes(y = logACI, x = Species)) +
  geom_violin(aes(fill = Species)) +
  geom_crossbar(stat="summary",
                fun = mean,
                fun.max = mean,
                fun.min = mean,
                fatten = 1.5, width = 0.8) +
  geom_point(color="black", alpha = 0.2,
             size=1.7, 
             position = position_jitter(w = 0.05))  +
  scale_fill_manual(labels=c('*M. laciniatus*', '*M. nasutus*', '*M. guttatus*'),
                    values= met.brewer("Tam", n = 4, type = "discrete", direction = -1)) +
  scale_color_manual(aes(color = "black")) +
  #scale_color_manual(labels=c(''*M. laciniatus*', '*M. nasutus*', '*M. guttatus*'),values= met.brewer("Archambault", n = 4, type = "discrete")) +
  theme_classic(base_size = 12) +
  theme(legend.title = element_text(face = "bold"),
        panel.border = element_rect(linetype = "blank", fill = NA),
        legend.text = element_markdown(size = 12), #Make legend text italic
        # axis.text.x = element_text(face="italic", 
        #                             size = 14,  
        #                             hjust = 1,
        #                             angle = 30)) +
        axis.text.x = element_blank()) +
  labs(y = expression(paste("log[ACI (% ", italic("A") [lambda]," at 931 nm / % ", italic("A") [lambda]," at 530 nm)]")), x = "") +
  #scale_x_discrete(labels = c('M. laciniatus', 'M. nasutus', 'M. guttatus')) +
  stat_compare_means(comparisons = my_comparisons,
                     method = "wilcox.test",
                     #label.y = c(0.4, 0.6, 1),
                     label = "p.signif") +
  facet_wrap(~Elevation_cat)
  #guides(color = "none")  +      #To turn off color legend
  #guides(fill = "none") + #To turn off fill legend
  #guides(shape = "none") + #To turn off shape legend
   
aci

#ggsave(filename = file.path(path, "figures/aci_violin.png"), plot = aci, dpi=300, units=c("mm"), width=300, height=150)
```

## Figure S3

```{r, supp_figS3}
#| echo: false
#| eval: true
#| tidy: true
#| fig-width: 6
#| fig-height: 6
#| label: supp_figS3
#| fig-cap: "Violin plots of Leaf Lobe Index (LBI) by species and elevation category."

#ggpubr package used for comparisons tests
lbi_compare_means <- compare_means(logLBI ~ Species,
         data = leaf_traits_noB,
         group.by = "Elevation_cat",
         method = "wilcox.test",
         p.adjust.method = "BH",
  paired = FALSE,
  alternative = "two.sided", 
  mu = 0, 
  var.equal = FALSE,
  conf.level = 0.95)

#lbi_compare_means


#Roughly the same outcome as pairwise_t_test.

#Report with `compare_means` because it is easier to plot with `ggpubr` package.
errbar_lims <- group_by(leaf_traits_noB, Species) |> 
              summarize(mean = mean(logLBI), se = sd(logLBI)/sqrt(n()), 
                        upper = mean + (2*se), lower = mean - (2*se))

#Comparisons
my_comparisons <- list(c("N", "L"), c("N", "G"), c("L", "G"))

lbi <- ggplot(leaf_traits_noB, aes(y = logLBI, x = Species)) +
  geom_violin(aes(fill = Species)) +
  #geom_boxplot(width = 0.3, color = "grey30", alpha = 0.2) +
  geom_crossbar(stat = "summary",
                fun = mean,
                fun.max = mean,
                fun.min = mean,
                fatten = 1.5, width = 0.8) +
  geom_point(color = "black", alpha = 0.2,
             size = 1.7,
             position = position_jitter(w = 0.05))  +
  scale_fill_manual(labels =c('*M. laciniatus*', '*M. nasutus*', '*M. guttatus*'),
                    values = met.brewer("Tam", n = 4, type = "discrete", direction = c(-1))) +
  scale_color_manual(aes(color = "black")) +
  #scale_color_manual(labels=c(''*M. laciniatus*', '*M. nasutus*', '*M. guttatus*'),values= met.brewer("Archambault", n = 4, type = "discrete")) +
  theme_classic(base_size = 12) +
  theme(legend.title = element_text(face = "bold"),
        panel.border = element_rect(linetype = "blank", fill = NA),
        legend.text = element_markdown(size = 12), #Make legend text italic
        axis.text.x = element_blank()) +
  labs(y = "log10[LBI (convex_hull - true_area]/convex_hull)]", x = "") +
  #scale_x_discrete(labels = c('M. laciniatus', 'M. nasutus', 'M. guttatus')) +
  stat_compare_means(comparisons = my_comparisons,
                     method = "wilcox.test",
                     #label.y = c(0.4, 0.6, 1),
                     label = "p.signif") +
   facet_wrap(~Elevation_cat)
 
  #guides(color = "none")  +      #To turn off color legend
  #guides(fill = "none") + #To turn off fill legend
  #guides(shape = "none") +#To turn off shape legend
  
lbi

ggsave(filename = file.path(path, "figures/lbi_violin.png"), plot = lbi, dpi=300, units=c("mm"), width=300, height=150)
```

## Figure S4

```{r, supp_figS4}
#| echo: false
#| eval: true
#| tidy: true
#| fig-width: 6
#| fig-height: 6
#| label: supp_figS4
#| fig-cap: "Violin plots of leaf toughness by species and elevation category."

#ggpubr package used for comparisons tests
lps_compare_means <- compare_means(logLPS ~ Species,
         data = leaf_traits_noB,
         group.by = "Elevation_cat",
         method = "wilcox.test",
         p.adjust.method = "BH",
  paired = FALSE,
  alternative = "two.sided", 
  mu = 0, 
  var.equal = FALSE,
  conf.level = 0.95)

#lps_compare_means

#Report with `compare_means` because it is easier to plot with `ggpubr` package.
errbar_lims <- group_by(leaf_traits_noB, Species) |> 
              summarize(mean = mean(logLPS), se = sd(logLPS)/sqrt(n()), 
                        upper = mean + (2*se), lower = mean - (2*se))

#Comparisons
my_comparisons <- list(c("N", "L"), c("N", "G"), c("L", "G"))

lps <- ggplot(leaf_traits_noB, aes(y = logLPS, x = Species)) +
  geom_violin(aes(fill = Species)) +
  #geom_boxplot(width = 0.3, color = "grey30", alpha = 0.2) +
  geom_crossbar(stat = "summary",
                fun = mean,
                fun.max = mean,
                fun.min = mean,
                fatten = 1.5, width = 0.8) +
  geom_point(color = "black", alpha = 0.2,
             size = 1.7,
             position = position_jitter(w = 0.05))  +
  scale_fill_manual(labels =c('*M. laciniatus*', '*M. nasutus*', '*M. guttatus*'),
                    values = met.brewer("Tam", n = 4, type = "discrete", direction = c(-1))) +
  scale_color_manual(aes(color = "black")) +
  #scale_color_manual(labels=c(''*M. laciniatus*', '*M. nasutus*', '*M. guttatus*'),values= met.brewer("Archambault", n = 4, type = "discrete")) +
  theme_classic(base_size = 12) +
  theme(legend.title = element_text(face = "bold"),
        panel.border = element_rect(linetype = "blank", fill = NA),
        legend.text = element_markdown(size = 12), #Make legend text italic
        axis.text.x = element_blank()) +
  labs(y = expression(paste("Leaf punch strength (N mm","^-1)")), x = "") +
  #scale_x_discrete(labels = c('M. laciniatus', 'M. nasutus', 'M. guttatus')) +
  stat_compare_means(comparisons = my_comparisons,
                     method = "wilcox.test",
                     #label.y = c(0.4, 0.6, 1),
                     label = "p.signif") +
   facet_wrap(~Elevation_cat)
 
  #guides(color = "none")  +      #To turn off color legend
  #guides(fill = "none") + #To turn off fill legend
  #guides(shape = "none") +#To turn off shape legend
  
lps


#ggsave(filename = file.path(path, "figures/lps_violin.png"), plot = lps, dpi=300, units=c("mm"), width=300, height=150)
```

## Figure S5

```{r, supp_figS5}
#| echo: false
#| eval: true
#| tidy: true
#| fig-width: 6
#| fig-height: 6
#| label: supp_figS5
#| fig-cap: "Violin plots of leaf thickness by species and elevation category."

#ggpubr package used for comparisons tests
lt_compare_means <- compare_means(logLT ~ Species,
         data = leaf_traits_noB,
         group.by = "Elevation_cat",
         method = "wilcox.test",
         p.adjust.method = "BH",
  paired = FALSE,
  alternative = "two.sided", 
  mu = 0, 
  var.equal = FALSE,
  conf.level = 0.95)

#lt_compare_means
#Roughly the same outcome as pairwise_t_test.

#Report with `compare_means` because it is easier to plot with `ggpubr` package.
errbar_lims <- group_by(leaf_traits_noB, Species) |> 
              summarize(mean = mean(logLT), se = sd(logLT)/sqrt(n()), 
                        upper = mean + (2*se), lower = mean - (2*se))

#Comparisons
my_comparisons <- list(c("N", "L"), c("N", "G"), c("L", "G"))

lt <- ggplot(leaf_traits_noB, aes(y= logLT, x = Species)) +
  geom_violin(aes(fill = Species)) +
  #geom_boxplot(width = 0.3, color = "grey30", alpha = 0.2) +
  geom_crossbar(stat = "summary",
                fun = mean,
                fun.max = mean,
                fun.min = mean,
                fatten = 1.5, width = 0.8) +
  geom_point(color = "black", alpha = 0.2,
             size = 1.7,
             position = position_jitter(w = 0.05))  +
  scale_fill_manual(labels =c('*M. laciniatus*', '*M. nasutus*', '*M. guttatus*'),
                    values = met.brewer("Tam", n = 4, type = "discrete", direction = c(-1))) +
  scale_color_manual(aes(color = "black")) +
  #scale_color_manual(labels=c(''*M. laciniatus*', '*M. nasutus*', '*M. guttatus*'),values= met.brewer("Archambault", n = 4, type = "discrete")) +
  theme_classic(base_size = 12) +
  theme(legend.title = element_text(face = "bold"),
        panel.border = element_rect(linetype = "blank", fill = NA),
        legend.text = element_markdown(size = 12), #Make legend text italic
        axis.text.x = element_blank()) +
        # axis.text.x = element_text(face="italic", 
        #                             size = 10,  
        #                             hjust = 1,
        #                             angle = 30)) +
  labs(y = expression(paste("Leaf thickness (",mu,"m)")), x = "") +
  scale_x_discrete(labels = c('M. laciniatus', 'M. nasutus', 'M. guttatus')) +
  stat_compare_means(comparisons = my_comparisons,
                     method = "wilcox.test",
                     #label.y = c(0.4, 0.6, 1),
                     label = "p.signif") +
   facet_wrap(~Elevation_cat)
 
  #guides(color = "none")  +      #To turn off color legend
  #guides(fill = "none") + #To turn off fill legend
  #guides(shape = "none") +#To turn off shape legend

lt

#ggsave(filename = file.path(path, "figures/lt_violin.png"), plot = lt, dpi=300, units=c("mm"), width=300, height=150)
```
